lib/internal/Magento/Framework/Css/PreProcessor/Instruction/MagentoImport.php

```diff
@@ -108,24 +134,51 @@ protected function replace(array $matchedContent, LocalInterface $asset)
             $relatedAsset = $this->assetRepo->createRelated($matchedFileId, $asset);
             $resolvedPath = $relatedAsset->getFilePath();
             $importFiles = $this->fileSource->getFiles($this->getTheme($relatedAsset), $resolvedPath);
+            $deployOnlyEnabled = $this->hasEnabledFlagDeployEnabledModules();
             /** @var $importFile \Magento\Framework\View\File */
             foreach ($importFiles as $importFile) {
+                $moduleName = $importFile->getModule();
                 $referenceString = $isReference ? '(reference) ' : '';
-                $importsContent .= $importFile->getModule()
-                    ? "@import $referenceString'{$importFile->getModule()}::{$resolvedPath}';\n"
-                    : "@import $referenceString'{$matchedFileId}';\n";
+
+                if (!$moduleName) {
+                    $importsContent .= "@import $referenceString'{$matchedFileId}';\n";
+                }
+
+                if ($moduleName) {
+                    if (!$deployOnlyEnabled) {
+                        $importsContent .= "@import $referenceString'{$moduleName}::{$resolvedPath}';\n";
+                    } else {
+                        if (!$this->moduleManager->isEnabled($moduleName)) {
+                            continue;
+                        }
+
+                        $importsContent .= "@import $referenceString'{$moduleName}::{$resolvedPath}';\n";
+
+                    }
+                }
```

Comment:
Maybe we can simplify to something like this?
```suggestion

                if ($moduleName) {
                    if (!$deployOnlyEnabled || $this->moduleManager->isEnabled($moduleName)) {
                        $importsContent .= "@import $referenceString'{$moduleName}::{$resolvedPath}';\n";
                    }
                } else {
                    $importsContent .= "@import $referenceString'{$matchedFileId}';\n";
                }
```